src/Microsoft.Performance.SDK/Processing/ColumnBuilding/ModalColumnBuilder.cs

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.Performance.SDK.Processing.ColumnBuilding;
+
+/// <summary>
+///     A builder for configuring modes on a column that has been
+///     configured as having mutually exclusive modes.
+/// </summary>
+public abstract class ModalColumnBuilder
+    : ColumnBuilder
+{
+    internal ModalColumnBuilder()
```

Comment:
>internal [](http://example.com/codeflow?start=4&length=8)

same here and for abstract classes in general: some sort of protected constructor - private protected in this case